libs/utils/src/sync/gate.rs

```diff
@@ -102,48 +128,81 @@ impl Gate {
     /// important that the holders of such guards are respecting a CancellationToken which has
     /// been cancelled before entering this function.
     pub async fn close(&self) {
-        warn_if_stuck(self.do_close(), &self.name, Duration::from_millis(1000)).await
+        let started_at = std::time::Instant::now();
+        let mut do_close = std::pin::pin!(self.do_close());
+
+        let nag_after = Duration::from_secs(1);
+        match tokio::time::timeout(nag_after, &mut do_close).await {
+            Ok(()) => return,
+            Err(_timeout) => {}
+        }
+
+        tracing::warn!(
```

Comment:
Unsure if this eneds to be `warn`, because we could have these in every test. When looking at a hang operation from logs, we would always notice this.

Reply:
Demoted to info on [03b71c9](https://github.com/neondatabase/neon/pull/6542/commits/03b71c9f788ae981d28cb9c6be0002553e13d002)